### PR TITLE
Deprecate funnel and pickfiles

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by stefanpenner** [comment]
@rwjblue r?

File: lib/models/addon.js [hunk 1]
```diff
@@ -112,14 +116,17 @@ Addon.prototype._requireBuildPackages = function() {
 
   this.transpileModules  = this.transpileModules || require('broccoli-es6modules');
```
Comment (rwjblue) [inline lib/models/addon.js]: Needs to be deprecated also

Comment (stefanpenner) [inline lib/models/addon.js]: good catch.

**Comment by stefanpenner** [comment]
@rwjblue updated

File: lib/models/addon.js [hunk 2]
```diff
@@ -103,24 +107,43 @@ Addon.prototype.hintingEnabled = function() {
   @private
   @method _requireBuildPackages
  */
+
 Addon.prototype._requireBuildPackages = function() {
```
Comment (rwjblue) [inline lib/models/addon.js]: Since this now does nothing should we deprecate it also?

Comment (stefanpenner) [inline lib/models/addon.js]: i need it actually to get good error messages.

Comment (rwjblue) [inline lib/models/addon.js]: Ok, I don't see that in this pull request.

Comment (stefanpenner) [inline lib/models/addon.js]: its updated.

Comment (stefanpenner) [inline lib/models/addon.js]: been running this on several apps at work, and working hard to make nice deprecation output (and catching other cases i didn't think of)

I think i got them all now

**Comment by samselikoff** [comment]
I changed this in my addon but consumers are having trouble. I tried adding `broccoli-merge-trees` and `broccoli-funnel` to devDependencies and dependencies, but if they used

```
npm install samselikoff/ember-cli-mirage
```

to install master, they get the error

> Cannot find module 'broccoli-merge-trees'

Any thoughts on how to fix? And, is it okay to use `this.Funnel` and `this.mergeTrees` for now?

**Comment by samselikoff** [comment]
actually, adding to `dependencies` did fix. Is this the suggested path? If so I can add some clarification in the docs

**Comment by stefanpenner** [comment]
Yup

**Comment by samselikoff** [comment]
What about a line [here](http://www.ember-cli.com/#packagejson) for `dependencies` that says 

```js
"dependencies": {
  // add npm packages your addon depends on here
```

? It could be me but I'm still not clear on what goes there vs. devDependencies. Basically if my addon uses an npm package in index.js, it needs to be in dependencies?

This absolutely could have nothing to do with ember-cli and is just my lack of understanding of npm

**Pull request merged**